src/Qir/Controller/Driver/QirDriverGenerator.cs

```diff
@@ -5,22 +5,48 @@
 using System.IO;
 using System.Threading.Tasks;
 using Microsoft.Quantum.Qir.Utility;
+using Microsoft.Quantum.QsCompiler;
 using Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint;
 
 namespace Microsoft.Quantum.Qir.Driver
 {
     public class QirDriverGenerator : IQirDriverGenerator
     {
+        private const string BytecodeFileName = "qir.bc";
+        private const string DriverFileName = "driver.cpp";
         private readonly ILogger logger;
 
         public QirDriverGenerator(ILogger logger)
         {
             this.logger = logger;
         }
 
-        public Task GenerateQirDriverCppAsync(DirectoryInfo sourceDirectory, EntryPointOperation entryPointOperation, ArraySegment<byte> bytecode)
+        public async Task GenerateQirDriverCppAsync(DirectoryInfo sourceDirectory, EntryPointOperation entryPointOperation, ArraySegment<byte> bytecode)
         {
-            throw new NotImplementedException();
+            // Wrapped in a task because DirectoryInfo operations are not asynchronous.
+            await Task.Run(async () =>
```

Comment:
I think that it would be ok to not run this through a task. The only advantage of using a task is that creating the directory and opening the files to write would run async, but those are not intensive enough operations to warrant a task in my opinion.

Reply:
I think it's generally best to do File IO asynchronously. But since this is not a multithreaded program, I suppose it's probably fine to do it synchronously.